File: Assignment/Abstraction/Expressions/ProgramNode.cs
using System;
using System.Collections.Generic;

namespace Assignment.Abstraction.Expressions
{
    internal class ProgramNode: INode
    {
        public ProgramNode()
        {
            this.Children = new List<INode>();
        }

        public ProgramNode(List<INode> nodes)
        {
            this.Children = nodes;
        }

        public List<INode> Children { get; }

        public string DisplayName()
        {
            return nameof(ProgramNode);
        }
    }
}

File: src/main/java/ce305/abstraction/expressions/ProgramNode.java
package ce305.abstraction.expressions;

import java.util.ArrayList;
import java.util.List;

import ce305.abstraction.INode;

public class ProgramNode implements INode {
    public final List<INode> children;

    public ProgramNode() {
        super();
        this.children = new ArrayList<INode>();
    }

    public ProgramNode(List<INode> children) {
        super();
        this.children = children;
    }

    @Override
    public String getDisplayName() {
        return ProgramNode.class.getSimpleName();
    }
}